vm/src/obj/objsequence.rs

```diff
@@ -137,7 +139,154 @@ impl<T: Clone> PySliceableSequence for Vec<T> {
     }
 }
 
-pub fn get_item(
+pub trait SequenceProtocol
+where
+    Self: Sized,
+{
+    fn get_elements(&self) -> Vec<PyObjectRef>;
+    fn as_object(&self) -> &PyObjectRef;
+    fn into_object(self) -> PyObjectRef;
```

Comment:
I think this could be split off into a separate trait, and we could have a generic `impl<T> ... for PyRef<T>`